.github/workflows/call-windows-unit-tests.yaml

```diff
@@ -0,0 +1,180 @@
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: ${{ matrix.config.arch }}
+
+      - name: Get gzip command w/ chocolatey
+        uses: crazy-max/ghaction-chocolatey@v3
+        with:
+          args: install gzip -y
+
+      # http://man7.org/linux/man-pages/man1/date.1.html
+      - name: Get Date
+        id: get-date
+        run: |
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Restore cached packages of vcpkg
+        id: cache-unit-test-vcpkg-sources
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            C:\vcpkg\packages
+          key: ${{ runner.os }}-${{ matrix.config.arch }}-wintest-vcpkg-${{ steps.get-date.outputs.date }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.config.arch }}-wintest-vcpkg-
+          enableCrossOsArchive: false
+
+      - name: Build openssl with vcpkg
+        run: |
+          C:\vcpkg\vcpkg install --recurse openssl --triplet ${{ matrix.config.vcpkg_triplet }}
+        shell: cmd
+
+      - name: Build libyaml with vcpkg
+        run: |
+          C:\vcpkg\vcpkg install --recurse libyaml --triplet ${{ matrix.config.vcpkg_triplet }}
+        shell: cmd
+
+      - name: Save packages of vcpkg
+        id: save-vcpkg-sources
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            C:\vcpkg\packages
+          key: ${{ steps.cache-unit-test-vcpkg-sources.outputs.cache-primary-key }}
+          enableCrossOsArchive: false
+
+      - name: run unit-test for Fluent Bit packages (only for x86 and x64)
+        run: |
+          cmake -G "NMake Makefiles" `
+            -D FLB_TESTS_INTERNAL=On `
+            -D FLB_NIGHTLY_BUILD='${{ inputs.unstable }}' `
+            -D OPENSSL_ROOT_DIR='${{ matrix.config.openssl_dir }}' `
+            ${{ matrix.config.cmake_additional_opt }} `
+            -D FLB_LIBYAML_DIR=C:\vcpkg\packages\libyaml_${{ matrix.config.vcpkg_triplet }} `
+            -D FLB_WITHOUT_flb-rt-out_elasticsearch=On `
+            -D FLB_WITHOUT_flb-rt-out_td=On `
+            -D FLB_WITHOUT_flb-rt-out_forward=On `
+            -D FLB_WITHOUT_flb-rt-in_disk=On `
+            -D FLB_WITHOUT_flb-rt-in_proc=On `
+            -D FLB_WITHOUT_flb-it-parser=On `
+            -D FLB_WITHOUT_flb-it-unit_sizes=On `
+            -D FLB_WITHOUT_flb-it-network=On `
+            -D FLB_WITHOUT_flb-it-pack=On `
+            -D FLB_WITHOUT_flb-it-signv4=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_ec2=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_http=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_profile=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_sts=On `
+            -D FLB_WITHOUT_flb-it-aws_util=On `
+            -D FLB_WITHOUT_flb-it-input_chunk=On `
+            ../
+          cmake --build .
+          ctest --build-run-dir $PWD --output-on-failure
```

Comment:
Do we need to quote PWD?

Reply:
Probably best though just in case that changes in the future for larger runners or whatever

Reply:
Got it. fair enough.